src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -1951,11 +1955,11 @@ private static int IndexOfEquivalent<TSyntaxNode>(SyntaxNode newNode, List<Synta
             return -1;
         }
 
-        private static List<SyntaxNode?>? GetAncestors(SyntaxNode? root, SyntaxNode node, Func<SyntaxNode, bool> nodeSelector)
+        private static List<SyntaxNode?>? GetAncestors(SyntaxNode? root, SyntaxNode? node, Func<SyntaxNode, bool> nodeSelector)
         {
             List<SyntaxNode?>? list = null;
 
-            while (node != root)
+            while (node is object && node != root)
```

Comment:
>node is object [](start = 19, length = 14)

This change seems unnecessary if the parameter can be declared `SyntaxNode` rather than `SyntaxNode?`.

Reply:
It's necessary because of the `node = node.Parent` statement at the end of the loop. That means the loop start can see a `null` value for `node`. 

---
In reply to: [362864810](https://github.com/dotnet/roslyn/pull/40699#discussion_r362864810) [](ancestors = 362864810)

Reply:
Won't we only hit the case where `current` is `null` when `root` is also `null`? In short, isn't `current != root` sufficient?

---
In reply to: [362886206](https://github.com/dotnet/roslyn/pull/40699#discussion_r362886206) [](ancestors = 362886206,362864810)